airbyte-workers/src/main/java/io/airbyte/workers/WorkerApp.java

```diff
@@ -220,23 +221,46 @@ private void registerConnectionManager(final WorkerFactory factory) {
   }
 
   private void registerSync(final WorkerFactory factory) {
-    final ReplicationActivityImpl replicationActivity = getReplicationActivityImpl(replicationWorkerConfigs, replicationProcessFactory);
-
-    final NormalizationActivityImpl normalizationActivity = getNormalizationActivityImpl(
-        defaultWorkerConfigs,
-        defaultProcessFactory);
-
-    final DbtTransformationActivityImpl dbtTransformationActivity = getDbtActivityImpl(
-        defaultWorkerConfigs,
-        defaultProcessFactory);
-
-    final PersistStateActivityImpl persistStateActivity = new PersistStateActivityImpl(statePersistence, featureFlags);
-
-    final Worker syncWorker = factory.newWorker(TemporalJobType.SYNC.name(), getWorkerOptions(maxWorkers.getMaxSyncWorkers()));
+    registerSyncDataPlaneWorkers(factory);
+    registerSyncControlPlaneWorkers(factory);
+  }
 
-    syncWorker.registerWorkflowImplementationTypes(SyncWorkflowImpl.class);
-    syncWorker.registerActivitiesImplementations(replicationActivity, normalizationActivity, dbtTransformationActivity, persistStateActivity);
+  /**
+   * Data Plane workers handle the subset of SyncWorkflow activity tasks that should run within a Data
+   * Plane.
+   */
```

Comment:
Don't we also want the CHECK workers to be moved to the new dataplanes?  E.g. If I'm testing that the source pod can reach my database, I only whitelisted the IPs from a certain (AWS maybe) datacenter.

Reply:
I talked about this with Davin recently - right now the plan is to keep the Multi-Cloud project scope as minimal as possible, so we plan to leave CHECK workers in the control plane until we have an explicit need to move them. We don't expect this to block any use cases until we offer hybrid Cloud plans, where a customer is deploying workers on their own data plane infrastructure but relying on Airbyte Cloud's control plane.

If we're wrong about that and this does become a blocker, we think it will be a small lift to enable CHECK workers to run in a data plane, as they don't have any specific database queries that will need conversion to new API endpoints or anything like that. It should just be a matter of tweaking our Temporal worker registration logic to respect EnvConfigs. 

@davinchia if I've missed any details, please clarify/correct as you see fit!

Reply:
That's right!